### DISCUSSION TITLE
hi, i had some issue for render text on the screen

### DISCUSSION THREAD

**Discussion by ChrisChou-freeman** [hajimehoshi/ebiten#1746] [Q&A] [answered]
Hello, I am a rookie who has only recently used Golang. When I try to render text on the screen using Ebiten, the font position will drift according to different font types instead of the default 0,0 default coordinates. I am very confused. Why is it so. Thank you for reading my question, if you know the answer please let me know.

**Comment by ModNope** [comment, marked as answer]
Drawing glyphs of the font starts from the bottom corner. If you do not take into account the size of the glyph, then at coordinates 0,0 the glyph will be off-screen at the top.

![image](https://user-images.githubusercontent.com/86792640/128631009-41eaf75c-6567-40dd-b908-0f7400eb0707.png)
After adding a vertical offset, the text will be below the screen border.

![image](https://user-images.githubusercontent.com/86792640/128631028-b1316435-5113-4b41-a37b-77aefd8a960e.png)

To account for the offset, there is a function that reports the length and width of the text with the selected font.

```go
Bound := BoundString(Font, Text)
// bound is image.Rectangle{}
// For get x, y length, use .Dx(), and .Dy()
// https://pkg.go.dev/image#Rectangle
```
If you need to display texts with different fonts in 1 zone at a fixed size, use 1 font as the standard size. 

```go

var(
	Default := // default font
	StandardZoneSize := BoundString(Default, Text)
)
...
func Drive(MyFont, text){
	if MyFont == nil{
		Draw(img, text, Default, x, y-StandardZoneSize.Dy(), color)
	}else{
		MyZoneSize := BoundString(MyFont, Text)
		Draw(img, text, MyFont, x, y-StandardZoneSize.Dy(), color) // use only standard size
	}
}
```

To place the text in the center, subtract its size from the size of the zone and divide by 2 - the size of the text, this is the offset for drawing.

```go
Xp := TextSize.Dx() - ( TextSize.Dx()-ZoneSize.Dx() )/2
//or
Xp := ZoneSize.Dx() + ( ZoneSize.Dx()-TextSize.Dx() )/2
//or
Xp := ( ZoneSize.Dx() + TextSize.Dx() )/2 // ~ lerp(ZoneSize.Dx(), TextSize.Dx(), 0.5)
```

**Reply by ChrisChou-freeman** [reply]
ok i got, thanks

**Reply by ModNope** [reply]
![image](https://user-images.githubusercontent.com/86792640/128632037-a8a5cc11-7817-45f2-8f63-001dd7cb2c88.png)
It's also worth adding that the text can be jagged due to the tops and bottoms of the glyphs. Because of this, it will be more correct to use an incomplete length for offset.

And the values
```go
Bound.Min.X, Y
Bound.Max.X, Y
```
These are 2 fields of type image.Point {}
and they contain extreme points. Values below 0 are front and bottom protrusions

**Reply by ChrisChou-freeman** [reply]
I learned it, thank you very much
